projects/deerfi/index.js

```diff
@@ -0,0 +1,59 @@
+/*** This Protocol seems hacked/rugged. It's a case for study and analyze ***/
+
+const sdk = require("@defillama/sdk");
+const abi = require("./abi.json");
+
+const factoryContract = "0xa22F8cf50D9827Daef24dCb5BAC92C147a9D342e";
+
+const ethTvl = async (timestamp, block) => {
+  const balances = {};
+
+  const lengthOfPools = (
+    await sdk.api.abi.call({
+      abi: abi.allPoolsLength,
+      target: factoryContract,
+      block,
+    })
+  ).output;
+
+  for (let i = 0; i < lengthOfPools; i++) {
+    const pool = (
+      await sdk.api.abi.call({
+        abi: abi.allPools,
+        target: factoryContract,
+        params: i,
+        block,
+      })
+    ).output;
+
+    const token = (
+      await sdk.api.abi.call({
+        abi: abi.token,
+        target: pool,
+        block,
+      })
+    ).output;
+
+    const reserve = (
+      await sdk.api.abi.call({
+        abi: abi.reserve,
+        target: pool,
+        block,
+      })
+    ).output;
+
+    sdk.util.sumSingleBalance(balances, token, reserve);
+  }
+
+  return balances;
+};
+
+module.exports = {
+  misrepresentedTokens: true,
+  ethereum: {
+    tvl: ethTvl,
+  },
+  tvl: sdk.util.sumChainTvls([ethTvl]),
```

Comment:
btw, if it's a tvl adapter (not a fetch one) there's no need to provide global tvl anymore, I made it so test.js automatically sums the tvl of each chain

Reply:
yeah, that's what i'm trying to do lastly